src/main/attached_database.cpp

```diff
@@ -69,7 +70,9 @@ AttachedDatabase::~AttachedDatabase() {
 			if (!config.options.checkpoint_on_shutdown) {
 				return;
 			}
-			storage->CreateCheckpoint(true);
+			// FIXME: is this super ugly???
```

Comment:
I'm not entirely sure what the implications of this are, CreateCheckpoint needed a client context, potentially this could be an `optional_ptr<ClientContext>` so we can pass `nullptr` here, but I believe I already tried that.

Needs another look though